bevm/api.go

```diff
@@ -664,6 +664,74 @@ func decodeEvmArray(abiType string, abi abi.Argument, arg interface{}) (
 	return arr.Interface(), nil
 }
 
+// EncodeEvmResult encodes an EVM call result into JSON.
+// This can be useful for command-line tools or calls serialized over the
+// network.
+func EncodeEvmResult(result interface{}, outputs abi.Arguments) (string, error) {
+	var jsonData []byte
+	var err error
+
+	if len(outputs) == 1 {
+		abiType := outputs[0].Type.String()
+		encodedResult, err := encodeEvmValue(abiType, result)
+		if err != nil {
+			return "", xerrors.Errorf("failed to encode EVM value: %v", err)
+		}
+
+		jsonData, err = json.Marshal(encodedResult)
+	} else {
+		resultArr, ok := result.([]interface{})
+		if !ok {
+			return "", xerrors.Errorf("received result type: %v, expected to "+
+				"be a slice", reflect.TypeOf(result))
+		}
+		if len(resultArr) != len(outputs) {
+			return "", xerrors.Errorf("received slice of length %v, expected "+
+				"to be of length %v", len(resultArr), len(outputs))
+		}
+
+		encodedResult := []interface{}{}
```

Comment:
If you `encodeResult := make([]interface{}, len(outputs))`, you can directly assign it on line 696...